src/lib/microservices/scanner-api/scanner-api.ts

```diff
@@ -0,0 +1,342 @@
+        throw e;
+    }
+
+    return {
+        config: databaseJob.config,
+        error: databaseJob.error,
+        finished: databaseJob.finished,
+        hints: databaseJob.hints,
+        id: databaseJob.id,
+        isNew: true,
+        maxRunTime: databaseJob.maxRunTime,
+        queued: databaseJob.queued,
+        started: databaseJob.started,
+        status: databaseJob.status,
+        url: databaseJob.url,
+        webhintVersion: null
+    };
+};
+
+/**
+ * Get the current active configuration.
+ */
+const getActiveConfig = async (): Promise<IServiceConfig> => {
+    let currentConfig: IServiceConfig;
+
+    try {
+        await database.connect(dbConnectionString);
+        currentConfig = await database.serviceConfig.getActive();
+
+    } catch (e) {
+        logger.error(`Could not connect to databse`, moduleName, e);
+        throw e;
+    }
+
+    if (!currentConfig) {
+        throw new Error('There is no active configuration');
+    }
+
+    const result: IServiceConfig = {
+        active: currentConfig.active,
+        jobCacheTime: currentConfig.jobCacheTime,
+        jobRunTime: currentConfig.jobRunTime,
+        name: currentConfig.name,
+        webhintConfigs: currentConfig.webhintConfigs
+    };
+
+    return result;
+};
+
+/** Create a job to scan an url if it doesn't exist. */
+export const createJob = async (url: string): Promise<IJob> => {
+    /*
+     *   1. Validate input data
+     *   2. Parse input data
+     *   3. Lock database by url
+     *   4. Check if the job exists having into account if the configuration is the same
+     *       a) If the job exists
+     *           I) The job is obsolete
+     *               i) Create a new job
+     *               ii) Add job to the queue
+     *           II) The job isn't obsolte => return existing job
+     *       b) If the job doesn't exist
+     *           I) Create a new job
+     *           II) Add job to the queue
+     *   5. Unlock database by url
+     */
+
+    if (!url) {
+        throw new Error('Url is required');
+    }
+
+    /*
+     * TODO
+     * We want to strip out the ability to upload configs in the future
+     * This just plugs in just the URL to the exsiting feature
+     * This will be refactored in a future PR to completely remove
+     * config upload feature
+     * https://github.com/webhintio/online-service/issues/585
+     */
+    const jobData: JobData = {
+        config: null,
+        hints: null,
+        source: ConfigSource.default,
+        url: url ? url : null
+    };
+
+    const serviceConfig: IServiceConfig = await getActiveConfig();
+
+    try {
+        await database.connect(dbConnectionString);
+    } catch (e) {
+        logger.error(`Could not connect to databse`, moduleName, e);
+        throw e;
+    }
+
+    const lock = await database.lock(jobData.url);
+
+    const config: Array<UserConfig> = getConfig(jobData, serviceConfig);
```

Comment:
`getConfig` is not used anymore, this should be just `const config = serviceConfig.webhintConfigs`
```suggestion
    const config = serviceConfig.webhintConfigs;
```